tests/discriminant.rs

```diff
@@ -1,177 +1,673 @@
-use std::intrinsics::discriminant_value;
+#[macro_use]
+mod util;
+
+use std::cmp::Ordering;
+
+use derive_where::derive_where;
+
+use self::util::Wrapper;
 
 #[test]
 fn default() {
+	#[derive_where(PartialEq, PartialOrd)]
 	enum Test {
-		A,
+		A = 0,
 		B,
+		#[allow(dead_code)]
+		#[derive_where(incomparable)]
 		C,
 	}
 
-	#[cfg(feature = "nightly")]
-	{
-		assert_eq!(0_isize, discriminant_value(&Test::A));
-		assert_eq!(1_isize, discriminant_value(&Test::B));
-		assert_eq!(2_isize, discriminant_value(&Test::C));
+	let test_a_1 = Test::A;
+	let test_a_2 = Test::A;
+
+	let test_b_1 = Test::B;
+	let test_b_2 = Test::B;
+
+	assert!(test_a_1 == test_a_2);
+	assert!(test_a_1 != test_b_1);
+	assert!(test_b_1 == test_b_2);
+	assert!(test_b_1 != test_a_1);
+
+	assert_eq!(test_a_1.partial_cmp(&test_a_2), Some(Ordering::Equal));
+	assert!(test_a_1 < test_b_1);
+	assert_eq!(test_b_1.partial_cmp(&test_b_2), Some(Ordering::Equal));
+	assert!(test_b_1 > test_a_1);
+}
+
+#[test]
+fn default_reverse() {
+	#[derive_where(PartialEq, PartialOrd)]
+	enum Test {
+		A = 2,
+		B = 1,
+		#[allow(dead_code)]
+		#[derive_where(incomparable)]
+		C = 0,
+	}
+
+	let test_a_1 = Test::A;
+	let test_a_2 = Test::A;
+
+	let test_b_1 = Test::B;
+	let test_b_2 = Test::B;
+
+	assert!(test_a_1 == test_a_2);
+	assert!(test_a_1 != test_b_1);
+	assert!(test_b_1 == test_b_2);
+	assert!(test_b_1 != test_a_1);
+
+	assert_eq!(test_a_1.partial_cmp(&test_a_2), Some(Ordering::Equal));
+	assert!(test_a_1 > test_b_1);
+	assert_eq!(test_b_1.partial_cmp(&test_b_2), Some(Ordering::Equal));
+	assert!(test_b_1 < test_a_1);
+}
+
+#[test]
+fn default_mix() {
+	#[derive_where(PartialEq, PartialOrd)]
+	enum Test {
+		A = 1,
+		B = 0,
+		C = 2,
+		#[allow(dead_code)]
+		#[derive_where(incomparable)]
+		D,
 	}
-	assert_eq!(0, Test::A as isize);
-	assert_eq!(1, Test::B as isize);
-	assert_eq!(2, Test::C as isize);
+
+	let test_a_1 = Test::A;
+	let test_a_2 = Test::A;
+
+	let test_b_1 = Test::B;
+	let test_b_2 = Test::B;
+
+	let test_c_1 = Test::C;
+	let test_c_2 = Test::C;
+
+	assert!(test_a_1 == test_a_2);
+	assert!(test_a_1 != test_b_1);
+	assert!(test_a_1 != test_c_1);
+	assert!(test_b_1 == test_b_2);
+	assert!(test_b_1 != test_a_1);
+	assert!(test_b_1 != test_c_1);
+	assert!(test_c_1 == test_c_2);
+	assert!(test_c_1 != test_a_1);
+	assert!(test_c_1 != test_b_1);
+
+	assert_eq!(test_a_1.partial_cmp(&test_a_2), Some(Ordering::Equal));
+	assert!(test_a_1 > test_b_1);
+	assert!(test_a_1 < test_c_1);
+	assert_eq!(test_b_1.partial_cmp(&test_b_2), Some(Ordering::Equal));
+	assert!(test_b_1 < test_a_1);
+	assert!(test_b_1 < test_c_1);
+	assert_eq!(test_c_1.partial_cmp(&test_c_2), Some(Ordering::Equal));
+	assert!(test_c_1 > test_a_1);
+	assert!(test_c_1 > test_b_1);
 }
 
 #[test]
-fn default_with_discriminants() {
+fn default_skip() {
+	#[derive_where(PartialEq, PartialOrd)]
 	enum Test {
-		A = 3,
+		A,
+		B = 3,
+		C,
+		#[allow(dead_code)]
+		#[derive_where(incomparable)]
+		E,
```

Comment:
that is not what I meant with "let's have an E"...

I meant
```suggestion
		#[allow(dead_code)]
		#[derive_where(incomparable)]
		D,
		E,
```

Reply:
:sweat_smile:

Reply:
Done.